packages/x-charts/src/models/seriesType/line.ts

```diff
@@ -33,6 +56,17 @@ export interface LineSeriesType
   area?: boolean;
   label?: string;
   curve?: CurveType;
+  /**
+   * Define which items o the series should display a mark.
+   * If can be a boolean that applies to all items.
+   * Or a callback that get some item properties and returns true if the item should be displayed.
+   */
+  showMark?: boolean | ((params: ShowMarkParams) => boolean);
```

Comment:
Have you considered inverting the boolean and naming it `hideMark`?
If I remember correctly, we usually try to default a boolean in such a way, that when you need it, it would be enough to add the property name in JSX, which would result in an implicit `true` value.

Reply:
Yes, but the rule does not really applies since it's an object property. We can not use  `{ hideMark }` to set `{ hideMark: true }`.

The reason I avoid this inversion is for the callback. It feels more natural to indicate where you want to display a mark than where you want to hide it. But that's more a feeling than a concert reason

Reply:
Well, as for the callback, in pickers we have `shouldDisable<...>` naming, but maybe it's not the best example as it refers to the validation. 🤔 
Naturally, the callback version would be most intuitive if it had a different name, like `shouldShowMark` or `shouldHideMark`, but that implies duplicating the property. 🙈 
Honestly, I don't have a strong opinion, the current solution seems to work and feel well. 👍